src/bin/multinode-demo.rs

```diff
@@ -196,26 +201,27 @@ fn mk_client(client_addr: &SocketAddr, r: &ReplicatedData) -> ThinClient {
     )
 }
 
-fn spy_node(client_addr: &SocketAddr) -> (ReplicatedData, UdpSocket) {
-    let mut addr = client_addr.clone();
-    addr.set_port(0);
-    let gossip = UdpSocket::bind(addr).unwrap();
+fn spy_node(client_addr: &Arc<RwLock<SocketAddr>>) -> (ReplicatedData, UdpSocket) {
+    let mut addr = client_addr.write().unwrap();
+    let port = addr.port();
+    let gossip = UdpSocket::bind(addr.clone()).unwrap();
+    addr.set_port(port + 1);
     let daddr = "0.0.0.0:0".parse().unwrap();
     let pubkey = KeyPair::new().pubkey();
     let node = ReplicatedData::new(pubkey, gossip.local_addr().unwrap(), daddr, daddr, daddr);
     (node, gossip)
 }
 
 fn converge(
-    client_addr: &SocketAddr,
+    client_addr: &Arc<RwLock<SocketAddr>>,
```

Comment:
These are some of the most tedious things to unwind. Are you sure you need all 3 of these, a ref, an Arc, and a RwLock?

Reply:
Yea.  We need all the client ports to be near the original so we can open up a range on aws